### PR TITLE
Add OVS to ControlPlane CR

### PR DESCRIPTION
**Pull request by slawqo**

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
+  newName: quay.io/skaplons/openstack-operator
```
Comment (dprince) [inline config/manager/kustomization.yaml:15]: Fixme: this should stay controller right?

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -5,18 +5,6 @@ metadata:
...
-  resources:
-  - cinders
-  verbs:
-  - create
-  - delete
-  - get
-  - list
-  - patch
-  - update
-  - watch
```
Comment (dprince) [inline config/rbac/role.yaml:19]: Did you mean to change the cinder permissions? perhaps a rebase is needed here?

File: apis/core/v1beta1/openstackcontrolplane_types.go [hunk 3]
```diff
@@ -175,6 +179,25 @@ type OvnResources struct {
...
 
+// OvsSection defines the desired state of OVS services
+type OvsSection struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=true
+	// Enabled - Whether OVS services should be deployed and managed
+	Enabled bool `json:"enabled,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Template - Overrides to use when creating the OVN services
```
Comment (karelyatin) [inline apis/core/v1beta1/openstackcontrolplane_types.go:190]: OVS

File: apis/core/v1beta1/openstackcontrolplane_types.go [hunk 4]
```diff
@@ -175,6 +179,25 @@ type OvnResources struct {
...
+	Template OvsResources `json:"template,omitempty"`
```
Comment (karelyatin) [inline apis/core/v1beta1/openstackcontrolplane_types.go:191]: can directly use Template ovsv1.OVSSpec `json:"template,omitempty"` to be consistent with other services(except ovn did a little differently as to manage both Northd and DBClusters).

File: pkg/openstack/ovs.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+			corev1beta1.OpenStackControlPlaneOVSReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackControlPlaneOVSReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	if op != controllerutil.OperationResultNone {
+		helper.GetLogger().Info(fmt.Sprintf("OVS %s - %s", ovs.Name, op))
+	}
```
Comment (karelyatin) [inline pkg/openstack/ovs.go:57]: Also need to set ready conditions like: https://github.com/openstack-k8s-operators/openstack-operator/blob/master/pkg/openstack/ovn.go#L101-L109

https://github.com/openstack-k8s-operators/ovs-operator/blob/main/api/v1beta1/ovs_types.go#L96

File: config/rbac/role.yaml (hunk 2 above)
Comment (karelyatin) [reply config/rbac/role.yaml:19]: +1 seems rebase missing + ovs permissions also missing as per
//+kubebuilder:rbac:groups=ovs.openstack.org,resources=ovs,verbs=get;list;watch;create;update;patch;delete
//+kubebuilder:rbac:groups=ovs.openstack.org,resources=ovs/status,verbs=get;update;patch

File: apis/go.mod [hunk 6]
```diff
@@ -48,6 +48,7 @@ require (
 	github.com/openshift/api v3.9.0+incompatible // indirect
 	github.com/openstack-k8s-operators/lib-common/modules/openstack v0.0.0-20220915080953-f73a201a1da6 // indirect
 	github.com/openstack-k8s-operators/lib-common/modules/storage v0.0.0-20220923094431-9fca0c85a9dc // indirect
+	github.com/openstack-k8s-operators/ovs-operator/api v0.0.0-20221201145301-12f408899b26
```
Comment (karelyatin) [inline apis/go.mod:51]: let's also update to latest https://github.com/openstack-k8s-operators/ovs-operator/commits/main

**Review by karelyatin — CHANGES_REQUESTED** [review]

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (slawqo) [reply config/manager/kustomization.yaml:15]: Correct

File: config/rbac/role.yaml (hunk 2 above)
Comment (slawqo) [reply config/rbac/role.yaml:19]: yes, sorry for that mistake. Fixed now

File: apis/core/v1beta1/openstackcontrolplane_types.go (hunk 4 above)
Comment (slawqo) [reply apis/core/v1beta1/openstackcontrolplane_types.go:191]: Done

File: apis/go.mod (hunk 6 above)
Comment (slawqo) [reply apis/go.mod:51]: Done

File: apis/core/v1beta1/openstackcontrolplane_types.go (hunk 3 above)
Comment (slawqo) [reply apis/core/v1beta1/openstackcontrolplane_types.go:190]: Done

File: pkg/openstack/ovs.go (hunk 5 above)
Comment (slawqo) [reply pkg/openstack/ovs.go:57]: Done

File: controllers/core/openstackcontrolplane_controller.go [hunk 7]
```diff
@@ -84,6 +85,8 @@ type OpenStackControlPlaneReconciler struct {
 //+kubebuilder:rbac:groups=neutron.openstack.org,resources=neutronapis,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=ovn.openstack.org,resources=ovndbclusters,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=ovn.openstack.org,resources=ovnnorthds,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ovs.openstack.org,resources=ovs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ovs.openstack.org,resources=ovs/status,verbs=get;update;patch
```
Comment (abays) [inline controllers/core/openstackcontrolplane_controller.go:89]: Does the OpenStack operator actually update the OVS CR status?  It doesn't seem like it does.  And I don't think OVS CR status should be updated outside of the OVS operator anyhow, as per best practices (perhaps).  We can probably remove this annotation.

Comment (slawqo) [reply controllers/core/openstackcontrolplane_controller.go]: Done

**Review by karelyatin — APPROVED** [review]
LGTM, worked fine in my tests !!

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**